hera_cal/redcal.py

```diff
@@ -1957,6 +1963,123 @@ def redcal_run(input_data, filetype='uvh5', firstcal_ext='.first.calfits', omnic
+    and write them out with that median.
+
+    Parameters
+    ----------
+    redcal_meta_file_list: list of str
+        list of file names containing redcal meta data. All files should have same number of times.
+
+    """
+    redcal_metas = []
+    for meta_file in redcal_meta_file_list:
+        redcal_metas.append((meta_file, ) + read_redcal_meta(meta_file))
+    # build list of delays and polarity flips for each antenna.
+    nfiles = len(redcal_meta_file_list)
+    ntimes = [len(rcm[4]) for rcm in redcal_metas]
+    delays = {ant: np.zeros(np.sum(ntimes)) for ant in redcal_metas[0][1]['dlys']}
+    offsets = {ant: np.zeros(np.sum(ntimes)) for ant in redcal_metas[0][1]['offsets']}
+    polarity_flips = {ant: copy.deepcopy(delays[ant]).astype(bool) for ant in delays}
+    nt = 0
+    for filenum, (meta_filename, fc_meta, omni_meta, freqs, times, lsts, antpos, history) in enumerate(redcal_metas):
+        for ant in fc_meta['dlys']:
+            tslice = slice(nt, nt + len(times))
+            delays[ant][tslice] = fc_meta['dlys'][ant]
+            polarity_flips[ant][tslice] = fc_meta['polarity_flips'][ant]
+            offsets[ant][tslice] = fc_meta['offsets'][ant]
+        nt += len(times)
+    # compute medians.
+    for ant in delays:
+        delays[ant] = np.nanmedian(delays[ant])
+        polarity_flips[ant] = bool(np.nanmedian(polarity_flips[ant]))
+        offsets[ant] = np.nanmedian(offsets[ant])
+
+    # update metadata and write out.
+    for filenum, (meta_filename, fc_meta, omni_meta, freqs, times, lsts, antpos, history) in enumerate(redcal_metas):
+        for ant in fc_meta['dlys']:
+            fc_meta['dlys'][ant][:] = delays[ant]
+            fc_meta['polarity_flips'][ant][:] = polarity_flips[ant]
+            fc_meta['offsets'][ant][:] = offsets[ant]
+        save_redcal_meta(meta_filename.replace(output_replace, output_ext), fc_meta, omni_meta, freqs, times, lsts, antpos,
+                         history + '\nTook nightly time median on delay and polarity flips.')
+
+
+def update_redcal_phase_degeneracy(redcal_file, redcal_meta_file, old_redcal_meta_file, output_file=None,
+                                   dont_replace_degens=False, dont_replace_polarities=False,
+                                   clobber=False):
+    """Update the phase degenerate component of a redcal solution and phase flips.
+
+    Parameters
+    ----------
+    redcal_file: str
+        path to calfits file containing redcal solution to update.
+    redcal_meta_file:
+        path to redcal meta file generated with io.save_redcal_meta()  with delays to be used to update degeneracy.
+    old_redcal_meta_file:
+        path to redcal meta file with old polarity flips.
+    output_file: str, optional
+        path to output file. Optional to write output to.
+    replace_degens: bool, optional
+        If True, replace degenerate portion of calibration solution with solution in redcal_meta_file.
+    fix_polarities: bool, optional
```

Comment:
If you have this option, it's weird that `old_redcal_meta_file` is a required parameter. Perhaps you want to put a way to allow the user to pass None for `old_redcal_meta_file` if this is False?